src/Storage/Bucket.php

```diff
@@ -81,15 +93,17 @@ class Bucket
      *        Storage.
      * @param string $name The bucket's name.
      * @param array $info [optional] The bucket's metadata.
+     * @param string $projectId The project ID.
      */
-    public function __construct(ConnectionInterface $connection, $name, array $info = [])
+    public function __construct(ConnectionInterface $connection, $name, array $info = [], $projectId)
```

Comment:
CI is failing, for a good reason, with the following error:
`Arguments with default values must be at the end of the argument list`

The `projectId` will be required in order to put together the full topic name in situations like [this example](https://github.com/GoogleCloudPlatform/google-cloud-php/pull/680/files#diff-61932b72d53dad17da23b15cb256b96bR538).

Moving the arguments around would be a breaking change, and making the `projectId` an optional argument either by passing it in via `info` or on its own could lead to problems if for some reason it isn't included. I've considered making the `projectId` accessible via the `connection` to circumvent this. Any thoughts on this subject would be greatly appreciated.

Reply:
After thinking about it overnight, I think adding a getter to the Connection is the best solution short of a BC-break. My vote is that, with the understanding that as soon as we have collected enough changes to justify a new major version we fix this along with that!